### ISSUE TITLE
Program stop with error message "cIn2Out::run: send on invalid socket"

### ISSUE THREAD

**Issue by JamesBremner** [JamesBremner/in2out#9]
Reported in #7

**Comment by JamesBremner** [comment]
Cause: Before  sending each line to output the code checks for a valid socket.  The server disconnect must happen in the brief moment between the check and the send. 

I will add handling for this exception.

**Comment by JamesBremner** [comment]
v1.4.2

**Comment by RmanSC** [comment]
Test was successful.
Test package (message), that was sent:
[21082022 message to sent.txt](https://github.com/JamesBremner/in2out/files/9403950/21082022.message.to.sent.txt)

Newest version 1.4.3 was tested on:
1. receiving data for 5 min without listener - success
2. receiving data for 5 min with listener - success
3. receiving data for 5 min after disconnection from listener - success
4. Only case, that make the app terminate with bad_alloc message was when I tried to turn on/off listener quickly to simulate bad connection. It happen to me only once. I tried to write better log file, but couldn't trigger that mistake and log of that occurance was erased already.
I think, it's minor issue and this issue can be closed.
